src/admin_extra/service_admin_extra_migration.erl

```diff
@@ -0,0 +1,788 @@
+%%% ==================================================================
+%%% @doc
+%%% MongooseIM migration
+%%% @end
+%%% ==================================================================
+
+-module(service_admin_extra_migration).
+
+-export([commands/0, migrate/3, get_all_data/1]).
+
+-export([
+  migrate_pubsub_nodes/1,
+  migrate_pubsub_affiliations/1,
+  migrate_pubsub_subscriptions/1,
+  migrate_pubsub_items/1,
+  migrate_vcard/1,
+  migrate_vcard_search/1,
+  migrate_users/1,
+  migrate_event_pusher_push_subscription/1,
+  migrate_rosterusers/1,
+  migrate_roster_version/1,
+  migrate_rostergroups/1,
+  migrate_last/1,
+  migrate_private_storage/1,
+  migrate_offline_message/1,
+  migrate_muc_light_rooms/1
+]).
+
+%%% ==================================================================
+%%% Records
+%%% ==================================================================
+
+-record(passwd, {us, password}).
```

Comment:
I'd like to avoid copying record definitions. This record is defined in `ejabberd_auth_mnesia`. I think the best would be to add functions returning data from the record to the `ejabberd_auth_mnesia` module and use them. Sth like this:

```erlang
get_us(#passwd{us = US}) -> US.
get_password(#passwd{password = Pass}) -> Pass.
```